backend/backend.js

```diff
@@ -30,8 +30,8 @@ app.get("/pins", async (req, res) => {
     res.send(result);
   } else if (pinType != undefined) {
     let result = await pinServices.filterByType(pinType);
-    if (result === undefined || result === null)
-      res.status(404).send("Resource not found.");
+    result = { pins_list: result };
```

Comment:
I think this might work better if placed after if(!result) check

Reply:
> 

Good catch, this should be resolved now